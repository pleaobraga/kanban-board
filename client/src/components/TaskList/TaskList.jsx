import './TaskList.scss'

import React from 'react'
import PropTypes from 'prop-types'
import { upperCase } from 'lodash'
import { Droppable } from 'react-beautiful-dnd'

import { default as Task } from '../Card'

export const TaskList = ({ id, name, Cards, index }) => {
  return (
    <div className="task-list">
      <h2 className="task-list__title">{upperCase(name)}</h2>
      <Droppable droppableId={id}>
        {(provided, snapshot) => (
          <div
            className={`task-list__cards-container ${snapshot.isDraggingOver &&
              'dragging-over'}`}
            ref={provided.innerRef}
            {...provided.droppableProps}
          >
            {Cards.map((card, cardIndex) => (
              <Task
                key={card.id}
                id={card.id}
                index={cardIndex}
                type={card.type}
                duration={card.duration}
                severity={card.severity}
                TaskListIndex={index}
              />
            ))}
            {provided.placeholder}
          </div>
        )}
      </Droppable>
    </div>
  )
}

TaskList.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  Cards: PropTypes.arrayOf(PropTypes.shape(Task.propTypes)),
  index: PropTypes.number.isRequired
}

export default TaskList
