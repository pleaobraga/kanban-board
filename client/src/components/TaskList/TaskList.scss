@import '../../styles/_color.scss';

.task-list {
  display: flex;
  flex-direction: column;
  box-sizing: border-box;

  padding: 1rem 0rem;

  border-left: 3px solid $dark-blue;
  border-right: 3px solid $dark-blue;

  max-width: 49rem;
  min-width: 30rem;

  min-height: 35rem;

  .dragging-over {
    background: skyblue;
  }

  &__title {
    font-family: 'Handlee', cursive;
    font-size: 5rem;

    text-align: center;
    width: 100%;

    border-bottom: 3px solid $dark-blue;
  }

  &__cards-container {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;

    height: 100%;

    padding: 2rem;

    .card {
      margin: 0.5rem;
    }
  }
}
