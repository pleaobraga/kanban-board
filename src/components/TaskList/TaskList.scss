@import '../../styles/_color.scss';

.task-list {
  display: flex;
  flex-direction: column;
  box-sizing: border-box;

  padding: 1rem 0rem;

  border-left: 3px solid $dark-blue;
  border-right: 3px solid $dark-blue;

  max-width: 49rem;

  .dragging-over {
    background: skyblue;
  }

  &--title {
    font-family: 'Handlee', cursive;
    font-size: 5rem;

    text-align: center;
    width: 100%;

    border-bottom: 3px solid $dark-blue;
  }

  &--cards-container {
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;

    padding: 2rem;

    .card {
      margin: 0.5rem;
    }
  }
}
