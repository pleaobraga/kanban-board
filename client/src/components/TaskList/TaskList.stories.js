import React from 'react'
import { storiesOf } from '@storybook/react'
import { text } from '@storybook/addon-knobs'
import { TaskList } from './TaskList'

const cards = [
  {
    id: '0',
    index: 0,
    type: 'feature',
    duration: 1,
    severity: 'hight'
  },
  {
    id: '1',
    index: 1,
    type: 'research',
    duration: 2,
    severity: 'medium'
  },
  {
    id: '2',
    index: 2,
    type: 'update',
    duration: 3,
    severity: 'low'
  }
]

export default storiesOf('Components | TaskList', module).add(
  'default',
  () => <TaskList name={text('name', 'Task List')} cards={cards} />,
  { info: { inline: true, header: false } }
)
