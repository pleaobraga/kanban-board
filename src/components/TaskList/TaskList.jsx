import './TaskList.scss'

import React from 'react'
import PropTypes from 'prop-types'
import { upperCase } from 'lodash'
import { Droppable } from 'react-beautiful-dnd'

import Card from '../Card'

export const TaskList = ({ id, title, cards }) => {
  return (
    <div className="task-list">
      <h2 className="task-list--title">{upperCase(title)}</h2>
      <Droppable droppableId={id}>
        {(provided, snapshot) => (
          <div
            className={`task-list--cards-container ${snapshot.isDraggingOver &&
              'dragging-over'}`}
            ref={provided.innerRef}
            {...provided.droppableProps}
          >
            {cards.map((card, index) => (
              <Card
                key={card.id}
                id={card.id}
                index={index}
                type={card.type}
                duration={card.duration}
                severity={card.severity}
              />
            ))}
            {provided.placeholder}
          </div>
        )}
      </Droppable>
    </div>
  )
}

TaskList.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  cards: PropTypes.arrayOf(
    PropTypes.shape({
      type: PropTypes.oneOf([
        'feature',
        'bugFix',
        'update',
        'research',
        'content'
      ]).isRequired,
      duration: PropTypes.number.isRequired,
      severity: PropTypes.oneOf(['hight', 'medium', 'low']).isRequired
    })
  )
}

export default TaskList
