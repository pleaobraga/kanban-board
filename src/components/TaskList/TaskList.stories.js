import React from 'react'
import { storiesOf } from '@storybook/react'
import { text } from '@storybook/addon-knobs'
import { TaskList } from './TaskList'

const cards = [
  {
    type: 'feature',
    duration: 1,
    severity: 'hight'
  },
  {
    type: 'research',
    duration: 2,
    severity: 'medium'
  },
  {
    type: 'update',
    duration: 3,
    severity: 'low'
  }
]

export default storiesOf('Components | TaskList', module).add(
  'default',
  () => <TaskList title={text('Title', 'Task List')} cards={cards} />,
  { info: { inline: true, header: false } }
)
