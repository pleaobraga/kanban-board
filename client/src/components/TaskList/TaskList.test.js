import React from 'react'
import { shallow } from 'enzyme'
import TaskList from './TaskList'

const cards = [
  {
    id: '0',
    index: 0,
    type: 'feature',
    duration: 1,
    severity: 'hight'
  },
  {
    id: '1',
    index: 1,
    type: 'research',
    duration: 2,
    severity: 'medium'
  },
  {
    id: '2',
    index: 2,
    type: 'update',
    duration: 3,
    severity: 'low'
  }
]

describe('Task List', () => {
  const component = shallow(<TaskList name="Task List" cards={cards} />)

  it('render properly', () => {
    expect(component).toMatchSnapshot()
  })
})
