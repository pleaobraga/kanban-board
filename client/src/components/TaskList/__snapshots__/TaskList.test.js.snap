// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task List render properly 1`] = `
<div
  className="task-list"
>
  <h2
    className="task-list--title"
  >
    TASK LIST
  </h2>
  <Connect(Droppable)
    direction="vertical"
    getContainerForClone={[Function]}
    ignoreContainerClipping={false}
    isCombineEnabled={false}
    isDropDisabled={false}
    mode="standard"
    renderClone={null}
    type="DEFAULT"
  >
    <Component />
  </Connect(Droppable)>
</div>
`;
